src/test/kotlin/se/uulm/snowballr/backend/service/criterion/CreateCriterionTest.kt

```diff
@@ -35,23 +66,117 @@ class CreateCriterionTest : MainServiceTest() {
+        coEvery { userRepoMock.getUserById(GrpcContext.getUserIdFromContext()) } returns user
+        coEvery { criterionRepoMock.createCriterion(any(), any()) } returns criterion
+        coEvery { projectRepoMock.getProjectById(project.id) } returns project
+        coEvery { projectMemberRepoMock.getAllProjectAdmins(project.id) } returns emptyList()
+
+        assertDoesNotThrow { mainService.createCriterion(request) }
     }
 
     @Test
-    fun `When a criterion is correctly created, then no exception is thrown`() = runTest {
-        val request = CriterionOuterClass.Criterion.Create.getDefaultInstance()
+    fun `When a project admin creates a project criterion, then no exception is thrown`() = runTest {
+        val user = DataBuilder.createExampleUser(id = dummyUserUUID, role = UserRole.USER_ROLE_DEFAULT)
+        val project = DataBuilder.createExampleProject()
+        val projectMember = DataBuilder.createExampleProjectMember(userId = dummyUserUUID, projectId = project.id)
+
+        val criterion = DataBuilder.createExampleCriterion(projectId = project.id)
+        val request = getProjectCriterionRequest(project.id.toString())
+
+        every { GrpcContext.getUserIdFromContext() } returns dummyUserUUID
+        coEvery { userRepoMock.getUserById(GrpcContext.getUserIdFromContext()) } returns user
+        coEvery { criterionRepoMock.createCriterion(any(), any()) } returns criterion
+
+        coEvery { projectRepoMock.getProjectById(project.id) } returns project
+        coEvery { projectMemberRepoMock.getAllProjectAdmins(project.id) } returns listOf(projectMember)
+
+        assertDoesNotThrow { mainService.createCriterion(request) }
+    }
+
+    @Test
+    fun `When a non project admin creates a project criterion, then an unauthorized exception is thrown`() = runTest {
+        val user = DataBuilder.createExampleUser()
+        val project = DataBuilder.createExampleProject()
+
+        val criterion = DataBuilder.createExampleCriterion(projectId = project.id)
+        val request = getProjectCriterionRequest(project.id.toString())
+
+        every { GrpcContext.getUserIdFromContext() } returns dummyUserUUID
+        coEvery { userRepoMock.getUserById(GrpcContext.getUserIdFromContext()) } returns user
+        coEvery { criterionRepoMock.createCriterion(any(), any()) } returns criterion
+        coEvery { projectRepoMock.getProjectById(project.id) } returns project
+        coEvery { projectMemberRepoMock.getAllProjectAdmins(project.id) } returns emptyList()
+        coEvery { criterionRepoMock.createCriterion(request, dummyUserUUID) } returns criterion
+
+        assertThrows<SnowballRException.UnauthorizedException.Create> { mainService.createCriterion(request) }
+    }
+
+    @ParameterizedTest
+    @CsvSource(
+        value = [
+            "PROJECT_STATUS_ARCHIVED",
+            "PROJECT_STATUS_ACTIVE_LOCKED",
+            "PROJECT_STATUS_DELETED",
+        ],
+    )
```

Comment:
I really like that you use parameterized testing here. For gRPC enums, we can use the `GrpcEnumSourceTest`, which can exclude enum values by names since some weeks.
```suggestion
@GrpcEnumSourceTest(ProjectOuterClass.ProjectStatus::class, excludes = ["PROJECT_STATUS_ACTIVE", "PROJECT_STATUS_UNSPECIFIED"])
```

We can also include `UNSPECIFIED` because it is also non-active in this case.

Reply:
That's a great suggestion, thanks.